examples/flower_classifier/image_pyfunc.py

```diff
@@ -0,0 +1,156 @@
+        self._column_names = ["predicted_label", "predicted_label_id"] + probs_names
+
+    def predict(self, input):
+        """
+        Generate predictions for the data.
+
+        :param input: pandas.DataFrame with one column containing images to be scored. The image
+                     column must contain base64 encoded binary content of the image files. The image
+                     format must be supported by PIL (e.g. jpeg or png).
+
+        :return: pandas.DataFrame containing predictions with the following schema:
+                     Predicted class: string,
+                     Predicted class index: int,
+                     Probability(class==0): float,
+                     ...,
+                     Probability(class==N): float,
+        """
+
+        # decode image bytes from base64 encoding
+        def decode_img(x):
+            return pd.Series(base64.decodebytes(bytearray(x[0], encoding="utf8")))
+
+        images = input.apply(axis=1, func=decode_img)
+        probs = self._predict_images(images)
+        m, n = probs.shape
+        label_idx = np.argmax(probs, axis=1)
+        labels = np.array([self._domain[i] for i in label_idx], dtype=np.str).reshape(m, 1)
+        output_data = np.concatenate((labels, label_idx.reshape(m, 1), probs), axis=1)
+        res = pd.DataFrame(columns=self._column_names, data=output_data)
+        res.index = input.index
+        return res
+
+    def _predict_images(self, images):
+        """
+        Generate predictions for input images.
+        :param images: binary image data
+        :return: predicted probabilities for each class
+        """
+
+        def preprocess_f(z):
+            return decode_and_resize_image(z, self._image_dims[:2])
+
+        x = np.array(
+            images[images.columns[0]].apply(preprocess_f).tolist())
+        with self._graph.as_default():
+            with self._session.as_default():
+                return self._model.predict(x)
+
+
+def log_model(keras_model, artifact_path, image_dims, domain):
+    """
+    Log a KerasImageClassifierPyfunc model as an MLflow artifact for the current run.
+
+    :param keras_model: Keras model to be saved.
+    :param artifact_path: Run-relative artifact path this model is to be saved to.
+    :param image_dims: Image dimensions the Keras model expects.
+    :param domain: Labels for the classes this model can predict.
+    """
+
+    with TempDir() as tmp:
+        data_path = tmp.path("image_model")
+        os.mkdir(data_path)
+        conf = {
+            "image_dims": "/".join(map(str, image_dims)),
+            "domain": "/".join(map(str, domain))
+        }
+        with open(os.path.join(data_path, "conf.yaml"), "w") as f:
+            yaml.safe_dump(conf, stream=f)
+        keras_path = os.path.join(data_path, "keras_model")
+        mlflow.keras.save_model(keras_model, path=keras_path)
+        conda_env = tmp.path("conda_env.yaml")
+        _mlflow_conda_env(
```

Comment:
This is a private function so it's kind of weird to use it in an example. Can we just write the environment as a big string?